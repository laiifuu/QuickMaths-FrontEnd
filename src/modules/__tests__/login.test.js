import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/configureStore';
import Login from '../user-sessions/login';

describe('Login', () => {
  test('Login page renders as expected', async () => {
    const snapHomepage = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
        ,
      </Provider>,
    )
      .toJSON();
    expect(snapHomepage).toMatchSnapshot();
  });
});
