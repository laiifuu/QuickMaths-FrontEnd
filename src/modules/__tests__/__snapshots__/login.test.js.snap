// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Login page renders as expected 1`] = `
Array [
  <section
    className="user-page flex"
  >
    <div>
      <h1>
        Welcome Back
      </h1>
    </div>
    <form
      action=""
      className="user-form flex"
    >
      <input
        id="username"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        required={true}
        type="input"
      />
      <div
        className="error"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <p />
      </div>
      <div
        className="error"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <p>
          Username field can not be empty
        </p>
      </div>
      <button
        className="session-btn"
        name="login"
        onClick={[Function]}
        type="button"
      >
        Log In
      </button>
      <a
        href="/user/signup"
        onClick={[Function]}
      >
        <p
          className="session-redirect"
        >
          <em>
            Not a member? Sign up...
          </em>
        </p>
      </a>
    </form>
  </section>,
  ",",
]
`;
